src/config/config.go

```diff
@@ -65,40 +69,59 @@ func ProcessArgs() *Config {
 	flag.StringVar(&ret.HTTPSProxy, "https-proxy", "", "A proxy URL to use for creating HTTPS connections outside the cluster")
 	flag.StringVar(&ret.NoProxy, "no-proxy", "", "A comma-separated list of destination domain names, domains, IP addresses, or other network CIDRs to exclude proxying")
 	flag.StringVar(&ret.ServiceIPs, "service-ips", "", "All IPs of assisted service node")
-	flag.StringVar(&ret.HighAvailabilityMode, "high-availability-mode", "Full", "high-availability expectations. default is 'Full', which represents the behavior in a \"normal\" cluster. Use 'None' for single-node deployment")
+	flag.StringVar(&ret.HighAvailabilityMode, "high-availability-mode", "", "high-availability expectations, \"Full\" which represents the behavior in a \"normal\" cluster. Use 'None' for single-node deployment. Leave this value as \"\" for workers as we do not care about HA mode for workers.")
 	flag.BoolVar(&ret.CheckClusterVersion, "check-cluster-version", false, "Do not monitor CVO")
 	flag.StringVar(&ret.MustGatherImage, "must-gather-image", "", "Custom must-gather image")
 	flag.Var(&ret.DisksToFormat, "format-disk", "Disk to format. Can be specified multiple times")
 
 	var installerArgs string
 	flag.StringVar(&installerArgs, "installer-args", "", "JSON array of additional coreos-installer arguments")
-
 	h := flag.Bool("help", false, "Help message")
 	flag.Parse()
 
+	if err := ret.SetInstallerArgs(installerArgs); err != nil {
+		printHelpAndExit(err)
+	}
+
+	if h != nil && *h {
+		printHelpAndExit(nil)
+	}
+
 	if ret.NoProxy != "" {
 		utils.SetNoProxyEnv(ret.NoProxy)
 	}
-	if h != nil && *h {
-		printHelpAndExit()
+
+	ret.SetDefaults()
+	if err := ret.Validate(); err != nil {
```

Comment:
What's the point?
The only thing that r.Validate() is validating is role == worker  and HighAvailabilityMode = none, and that can't happen anymore due to SetDefaults.